internal/query/gin_map.go

```diff
@@ -2,19 +2,18 @@ package query
 
 import "github.com/gin-gonic/gin"
 
-// QueryNestedMap returns a map for a given query key.
+//revive:disable:exported We want to mimic the gin API.
+
+// ShouldGetQueryNestedMap returns a map from query params.
 // In contrast to QueryMap it handles nesting in query maps like key[foo][bar]=value.
-//
-//revive:disable:exported We want to mimic the gin API
-func QueryNestedMap(c *gin.Context, key string) (dicts map[string]interface{}) {
-	dicts, _ = GetQueryNestedMap(c, key)
-	return
+func ShouldGetQueryNestedMap(c *gin.Context) (dicts map[string]interface{}, err error) {
+	return ShouldGetQueryNestedMapForKey(c, "")
 }
 
-// GetQueryNestedMap returns a map for a given query key, plus a boolean value
-// whether at least one value exists for the given key.
-// In contrast to GetQueryMap it handles nesting in query maps like key[foo][bar]=value.
-func GetQueryNestedMap(c *gin.Context, key string) (map[string]interface{}, bool) {
+// ShouldGetQueryNestedMapForKey returns a map from query params for a given query key.
+// In contrast to QueryMap it handles nesting in query maps like key[foo][bar]=value.
+// Similar to ShouldGetQueryNestedMap but it returns only the map for the given key.
+func ShouldGetQueryNestedMapForKey(c *gin.Context, key string) (dicts map[string]interface{}, err error) {
```

Comment:
I suppose we should put these examples with additional description to our documentation.

Reply:
I created JIRA issue for updating documentation about search endpoints https://bitcoinassociation.atlassian.net/browse/SPV-1014